File: Modules/MovieDB/MovieCast/MovieCastForm.cs
using Serenity.ComponentModel;

namespace MovieTutorial.MovieDB.Forms;

[FormScript("MovieDB.MovieCast")]
[BasedOnRow(typeof(MovieCastRow), CheckNames = true)]
public class MovieCastForm
{
   // public int MovieId { get; set; }
    public int PersonId { get; set; }
    public string Character { get; set; }
  //  public string Character { get; set; }
}

File: Modules/ServerTypes/MovieDB/MovieCastForm.ts
import { LookupEditor, StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface MovieCastForm {
    PersonId: LookupEditor;
    Character: StringEditor;
}

export class MovieCastForm extends PrefixedContext {
    static readonly formKey = 'MovieDB.MovieCast';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MovieCastForm.init)  {
            MovieCastForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;

            initFormType(MovieCastForm, [
                'PersonId', w0,
                'Character', w1
            ]);
        }
    }
}